song-server/src/main/java/org/icgc/dcc/song/server/service/AnalysisService.java

```diff
@@ -301,18 +284,81 @@ VariantCall readVariantCall(String id) {
     return ok("AnalysisId %s was suppressed",id);
   }
 
+  public List<CompositeEntity> readSamples(String id) {
+    val samples = repository.findSampleIds(id).stream()
+        .map(compositeEntityService::read)
+        .collect(toImmutableList());
+
+    // If there are no samples, check that the analysis even exists. If it does, then the analysis is corrupted since
+    // it is missing samples
+    if (samples.isEmpty()){
+      checkAnalysis(id);
+      throw buildServerException(getClass(), ANALYSIS_MISSING_SAMPLES,
+          "The analysis with analysisId '%s' is missing samples and is therefore corrupted. "
+          + "It should map to at least 1 sample",id);
+    }
+    return samples;
+  }
+
+  private void saveCompositeEntities(String studyId, String id, List<CompositeEntity> samples) {
+    samples.stream()
+        .map(sample->compositeEntityService.save(studyId,sample))
+        .forEach(sampleId->repository.addSample(id, sampleId));
+  }
+
+  private void saveFiles(String id, String studyId, List<File> files) {
+    files.forEach(f->fileService.save(id, studyId, f));
```

Comment:
Just a question, is there anyway this can return a value like num of inserts or something like that?

Reply:
@andricDu since the `FileService.save` method returns the `objectId`, can the `AnalysisService.saveFiles` method return a `List<String>` of `objectId`s?